bindings/rabbitmq/rabbitmq_test.go

```diff
@@ -6,21 +6,266 @@
+		},
+		{
+			name:                     "Not Delete / Not Durable",
+			properties:               map[string]string{"QueueName": queueName, "Host": host, "DeleteWhenUnused": "false", "Durable": "false"},
+			expectedDeleteWhenUnused: false,
+			expectedDurable:          false,
+		},
+		{
+			name:                     "TTL",
+			properties:               map[string]string{"QueueName": queueName, "Host": host, "DeleteWhenUnused": "false", "Durable": "false", "ttl": "1"},
+			expectedDeleteWhenUnused: false,
+			expectedDurable:          false,
+			expectedTTL:              &oneSecondTTL,
+		},
+		{
+			name:                     "Empty TTL",
+			properties:               map[string]string{"QueueName": queueName, "Host": host, "DeleteWhenUnused": "false", "Durable": "false", "ttl": ""},
+			expectedDeleteWhenUnused: false,
+			expectedDurable:          false,
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			m := bindings.Metadata{}
+			m.Properties = tt.properties
+			r := RabbitMQ{logger: logger.NewLogger("test")}
+			err := r.parseMetadata(m)
+			assert.Nil(t, err)
+			assert.Equal(t, queueName, r.metadata.QueueName)
+			assert.Equal(t, host, r.metadata.Host)
+			assert.Equal(t, tt.expectedDeleteWhenUnused, r.metadata.DeleteWhenUnused)
+			assert.Equal(t, tt.expectedDurable, r.metadata.Durable)
+			assert.Equal(t, tt.expectedTTL, r.defaultQueueTTL)
+		})
+	}
+}
+
+func TestParseMetadataWithInvalidTTL(t *testing.T) {
+	const queueName = "test-queue"
+	const host = "test-host"
+
+	testCases := []struct {
+		name       string
+		properties map[string]string
+	}{
+		{
+			name:       "Whitespaces TTL",
+			properties: map[string]string{"QueueName": queueName, "Host": host, "ttl": "  "},
+		},
+		{
+			name:       "Negative ttl",
+			properties: map[string]string{"QueueName": queueName, "Host": host, "ttl": "-1"},
+		},
+		{
+			name:       "Non-numeric ttl",
+			properties: map[string]string{"QueueName": queueName, "Host": host, "ttl": "abc"},
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			m := bindings.Metadata{}
+			m.Properties = tt.properties
+			r := RabbitMQ{logger: logger.NewLogger("test")}
+			err := r.parseMetadata(m)
+			assert.NotNil(t, err)
+		})
+	}
+}
+
+func getMessageWithRetries(ch *amqp.Channel, queueName string, maxDuration time.Duration) (msg amqp.Delivery, ok bool, err error) {
+	start := time.Now()
+	for time.Since(start) < maxDuration {
+		msg, ok, err := ch.Get(queueName, true)
+		if err != nil || ok {
+			return msg, ok, err
+		}
+
+		time.Sleep(100 * time.Millisecond)
+	}
+
+	return amqp.Delivery{}, false, nil
+}
+
+func TestQueuesWithTTL(t *testing.T) {
+	rabbitmqHost := getTestRabbitMQHost()
+	if rabbitmqHost == "" {
+		t.Skipf("RabbitMQ binding integration tests skipped. To enable define the connection string using environment variable '%s' (example 'amqp://guest:guest@localhost:5672/')", testRabbitMQHostEnvKey)
```

Comment:
@youngbupark Could you please suggest whats the correct way to add and run these integration tests so that they are not included in default run.

Reply:
Sure. 

@fbeltrao Can we move all integration test codes to `rabbitmq_integration_test.go` ?

Please use `integration_test` build tag to separate the integration test from unit-test.

```
// +build integration_test

// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
// ------------------------------------------------------------

package rabbitmq

import (
	"encoding/json"
	"fmt"
	"os"
	"testing"

	"github.com/stretchr/testify/require"
)

func TestQueuesWithTTL(t *testing.T) {
...
}
...
```

Then you can run the integration test with the below command:

```
go test -v -tags=integration_test ./bindings/rabbitmq/...
```

Reply:
Thanks @youngbupark for the help!

Reply:
@youngbupark 
To run RabbitMQ tests I would need to start the broker in docker.
To run Service Bus integration tests I would need a SB connection string.

How do we provide it? Do we have docker in all build agents?

Reply:
@fbeltrao The test would still need to be run after setting env vars. This tag skips these when tests are run without any tags by doing: `go test`

Reply:
Done. I am not sure the tests are being executed though as you would have to start RabbitMQ on docker and have a Service Bus account.

Reply:
@fbeltrao Currently these integration  tests are not run as part  of CI and needs to be run manually. The work items for tests overall is tracked by this issue https://github.com/dapr/components-contrib/issues/204